runner/src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -305,6 +306,68 @@ export class PageControllerBase {
     return request.yar.get("lang");
   }
 
+  calculateRouteState(page, completeState, state = {}) {
+    //Store our current routeState outside of our recursive method where it will be appended during the recursion
+    this.routeState = state;
+
+    //From the page we passed in, iterate through the possible nextPages
+    page.pageDef.next.forEach((nextPage) => {
```

Comment:
If we can pass the condition checks required to get us there. ie the user has previously followed the form the whole way through. 

If we don't pass a condition check needed to get us there, we don't go any further since we have no applicable answers beyond that point. Hopefully that makes sense.